client/components/count/index.jsx

```diff
@@ -2,22 +2,22 @@
  * External dependencies
  */
 import React from 'react';
-import PureRenderMixin from 'react-pure-render/mixin';
+import { localize } from 'i18n-calypso';
+import { identity, omit } from 'lodash';
 
-export default React.createClass( {
+const Count = ( { count, numberFormat, ...inheritProps } ) => (
+	<span className="count" { ...omit( inheritProps, [ 'translate', 'moment' ] ) }>
+		{ numberFormat( count ) }
+	</span>
+);
 
-	displayName: 'Count',
+Count.propTypes = {
+	count: React.PropTypes.number.isRequired,
+	numberFormat: React.PropTypes.func
+};
 
-	mixins: [ PureRenderMixin ],
+Count.defaultProps = {
+	numberFormat: identity
```

Comment:
I suggested this, but I'm not entirely sure we actually need it. Did you ever check if it worked without on the server? We could try without for now and add it as part of #10515 if we discover we need it. (As for tests, we can just pass `identity` as a prop there.)